src/middlewared/middlewared/plugins/enclosure_/identification.py

```diff
@@ -1,46 +1,42 @@
-def r20_variants_or_mini(model, dmi):
-    """If this is an r20 variant then we return the model
-    with the TRUENAS- suffix stripped. Otherwise, if this is
-    a MINI then we just return the entire string.
-
-    NOTE: this information is burned in by the production team
-    into the motherboard (SMBIOS) before we ship the system
-    """
-    if dmi in ('TRUENAS-R20', 'TRUENAS-R20A', 'TRUENAS-R20B'):
-        return model, True
-    elif dmi.startswith(('TRUENAS-MINI', 'FREENAS-MINI')):
-        # minis do not have the TRUENAS- prefix removed
-        return dmi, True
-    else:
-        return '', False
-
-
 def get_enclosure_model_and_controller(key, dmi):
     """This maps the enclosure to the respective platform.
     the 'key' is the concatenated string of t10 vendor and product
     info returned by a standard INQUIRY command to the enclosure
     device.
     """
-    model = dmi.removeprefix('TRUENAS-').removesuffix('-HA')
+    model = dmi.removeprefix('TRUENAS-').removeprefix('FREENAS-')
+    model = model.removesuffix('-HA').removesuffix('-S')
     match key:
         case 'ECStream_4024Sp' | 'ECStream_4024Ss' | 'iX_4024Sp' | 'iX_4024Ss':
             # M series
             return model, True
         case 'CELESTIC_P3215-O' | 'CELESTIC_P3217-B':
             # X series
             return model, True
+        case 'BROADCOM_VirtualSES':
+            # H series
+            if dmi.startswith('TRUENAS-H'):
+                return model, True
+            else:
+                return '', False
         case 'ECStream_FS1' | 'ECStream_FS2' | 'ECStream_DSS212Sp' | 'ECStream_DSS212Ss':
             # R series
             return model, True
-        case 'iX_FS1' | 'iX_FS2' | 'iX_DSS212Sp' | 'iX_DSS212Ss':
+        case 'iX_FS1L' | 'iX_FS2' | 'iX_DSS212Sp' | 'iX_DSS212Ss':
             # more R series
             return model, True
-        case 'iX_TrueNAS R20p' | 'iX_TrueNAS 2012Sp' | 'iX_TrueNAS SMC SC826-P':
+        case 'iX_TrueNASR20p' | 'iX_2012Sp' | 'iX_TrueNASSMCSC826-P':
```

Comment:
Doesn't really have to be fixed here, but we need to consider creating / using an enum for all these strings. Looks like it would be quite easy to typo some of them somewhere.

Reply:
I can't use a traditional python enum for these strings. It's a 1-to-many relationship. (i.e. ES24 enclosure has 2 different strings that map back to it. Enums won't work for this) Furthermore, they're only used in this function.